Modify configure-blacksmith to include plan parameters

norman-abramovitz opened the pull request:
Changed configure_blacksmith to include the client parameters when the merge meta file is generated.

Commit:
Modify configure-blacksmith to include plan parameters
Modify plan parameter names per Tom's request

Commit:
add processing to disable lua scripting.  The default is enabled.

Commit:
Fixed default value setting for manifest files

Commit:
Trying to get boolean values to pass through correctly

Commit:
added maxmemory that is settable from the plan
plans can now set create-service parameters

Commit:
Hard coded that redis commmands DEBUG CONFIG SHUTDOWN SLAVEOF and SYNC
are disabled.

Commit:
Added in some append only file configuration parameters for use in plan
definitions.

Commit:
Fixed formatting

Commit:
Fixed typo

Commit:
Documented the new parameters.   This is work in progress.

thomasmitchell commented on README.md:
```diff
@@ -231,8 +231,48 @@ instance_groups:
+## Redis Hard Coded Redis Configuration Parameters
+
+For security reasons the following Redis commands have been disabled in all plan types. 
+This avoids Redis users from overriding the predefined Redis plan parameters.
```
I would just cut out line 237. 236 suffices.

norman-abramovitz replied:
done

thomasmitchell commented on jobs/cluster/templates/config/redis.conf:
```diff
@@ -16,12 +16,37 @@ appendonly yes
+auto-aof-rewrite-percentage "<%= p('redis.auto-aof-rewrite-percentage') %>"
+<% end %>
+<% if p('redis.auto-aof-rewrite-min-size') %>
+auto-aof-rewrite-min-size "<%= p('redis.auto-aof-rewrite-min-size') %>"
```
don't quote this

norman-abramovitz replied:
Interesting quotes did not cause any issues, but I will make the change.

thomasmitchell commented on jobs/cluster/templates/config/redis.conf:
```diff
@@ -16,12 +16,37 @@ appendonly yes
+no-appendfsync-on-rewrite "<%= p('redis.no-appendfsync-on-rewrite') %>"
+<% end %>
+<% if p('redis.auto-aof-rewrite-percentage') %>
+auto-aof-rewrite-percentage "<%= p('redis.auto-aof-rewrite-percentage') %>"
```
don't quote this

thomasmitchell commented on jobs/cluster/spec:
```diff
@@ -76,3 +94,15 @@ properties:
+
+  redis.no-appendfsync-on-rewrite:
+    description: If you have latency problems turn this to "yes". Otherwise leave it as "no"
+    default: "no"
```
let's make this a true/false boolean in the YAML. Especially because there's a potential gotcha here, as certain versions of YAML will interpret a non-quoted `yes` as the boolean `true` and an unquoted `no` as the boolean `false`.

thomasmitchell commented on jobs/redis-blacksmith-plans/templates/bin/configure-blacksmith:
```diff
@@ -40,6 +40,52 @@ meta:
+  redis_slowlog-max-len: <%= plan["redis_slowlog-max-len"] %>
+  <% end %>
+
+  <% if plan["redis_no-appendfsync-on-rewrite"] %>
```
let's turn this key into a boolean in yaml as well. This line is still fine after it's turned into a boolean though.

thomasmitchell commented on jobs/redis-blacksmith-plans/templates/bin/configure-blacksmith:
```diff
@@ -40,6 +40,52 @@ meta:
+  <% end %>
+
+  <% if plan["redis_no-appendfsync-on-rewrite"] %>
+  redis_no-appendfsync-on-rewrite <%= plan["redis_no-appendfsync-on-rewrite"] %>
```
You're missing a colon here to make it a YAML key/value. Also, when this is changed to a boolean, the ERB value should be changed to a ternary statement like 
```erb
<%= plan["redis_no-appendfsync-on-rewrite"] ? "yes" : "no" %>
```

thomasmitchell commented on jobs/redis-blacksmith-plans/templates/bin/configure-blacksmith:
```diff
@@ -40,6 +40,52 @@ meta:
+  <% if plan["client_connections"] %>
+  client_connections: <%= plan["client_connections"] %>
+  <% end %>
+  <% if plan["lua_scripting_enabled"] %>
+  lua_scripting_enabled: true
+  <% else %>
+  lua_scripting_enabled: false
+  <% end %>
```
more concisely:
```suggestion
  lua_scripting_enabled: <%= !!plan["lua_scripting_enabled"] %>
```

thomasmitchell commented on jobs/redis-blacksmith-plans/templates/plans/cluster/manifest.yml:
```diff
@@ -41,11 +41,17 @@ instance_groups:
+            slowlog-log-slower-than: (( grab meta.params.slowlog-log-slower-than || meta.redis_slowlog-log-slower-than || 10000 ))
+            slowlog-max-len: (( grab meta.params.slowlog-max-len || meta.redis_slowlog-max-len || 128 ))
+            maxmemory: (( grab meta.redis_maxmemory || 0 ))
+            no-appendfsync-on-rewrite: (( grab meta.redis_no-appendfsync-on-rewrite || "no" ))
```
if this is turned into a boolean, then this becomes
```
no-appendfsync-on-rewrite: (( grab meta.redis_no-appendfsync-on-rewrite || false ))
```

thomasmitchell commented on jobs/redis-blacksmith-plans/templates/plans/standalone/manifest.yml:
```diff
@@ -37,11 +37,17 @@ instance_groups:
+            slowlog-log-slower-than: (( grab meta.params.slowlog-log-slower-than || meta.redis_slowlog-log-slower-than || 10000 ))
+            slowlog-max-len: (( grab meta.params.slowlog-max-len || meta.redis_slowlog-max-len || 128 ))
+            maxmemory: (( grab meta.redis_maxmemory || 0 ))
+            no-appendfsync-on-rewrite: (( grab meta.redis_no-appendfsync-on-rewrite || "no" ))
```
if this is turned into a boolean, then this becomes
```
no-appendfsync-on-rewrite: (( grab meta.redis_no-appendfsync-on-rewrite || false ))
```

thomasmitchell commented on jobs/standalone/spec:
```diff
@@ -56,3 +74,15 @@ properties:
     description: Sets the length (count) of the slowlog queue.
     default: 128
+
+  redis.no-appendfsync-on-rewrite:
```
as has been said a bunch above, let's make this a YAML boolean.

thomasmitchell commented on jobs/standalone/templates/config/redis.conf:
```diff
@@ -7,13 +7,40 @@ appendonly yes
 appendfilename "redis.aof"
 appendfsync everysec
+
+<% if p('redis.no-appendfsync-on-rewrite') %>
+no-appendfsync-on-rewrite "<%= p('redis.no-appendfsync-on-rewrite') %>"
+<% end %>
```
once this is changed to a boolean, simply
```
no-appendfsync-on-rewrite <%= p('redis.no-appendfsync-on-rewrite') %> ? "yes" : "no" %>
```

thomasmitchell commented on jobs/standalone/templates/config/redis.conf:
```diff
@@ -7,13 +7,40 @@ appendonly yes
+no-appendfsync-on-rewrite "<%= p('redis.no-appendfsync-on-rewrite') %>"
+<% end %>
+<% if p('redis.auto-aof-rewrite-percentage') %>
+auto-aof-rewrite-percentage "<%= p('redis.auto-aof-rewrite-percentage') %>"
```
don't quote the erb value

thomasmitchell commented on jobs/standalone/templates/config/redis.conf:
```diff
@@ -7,13 +7,40 @@ appendonly yes
+auto-aof-rewrite-percentage "<%= p('redis.auto-aof-rewrite-percentage') %>"
+<% end %>
+<% if p('redis.auto-aof-rewrite-min-size') %>
+auto-aof-rewrite-min-size "<%= p('redis.auto-aof-rewrite-min-size') %>"
```
don't quote the erb value

norman-abramovitz replied:
fixed

thomasmitchell commented on jobs/standalone/templates/config/redis.conf:
```diff
@@ -7,13 +7,40 @@ appendonly yes
 timeout <%= p('client.timeout') %>
 tcp-keepalive <%= p('client.tcpkeepalive') %>
 maxclients  <%= p('client.connections') %>
+
+# Hard coding out these values per LM request
+
```
drop this comment. It's not really by request. This is a reasonable boundary for user of a service vs operator of a service

thomasmitchell commented on jobs/cluster/templates/config/redis.conf:
```diff
@@ -16,12 +16,37 @@ appendonly yes
 # we should already be *in* the store dir
 appendfilename "redis.aof"
 appendfsync everysec
+<% if p('redis.no-appendfsync-on-rewrite') %>
+no-appendfsync-on-rewrite "<%= p('redis.no-appendfsync-on-rewrite') %>"
+<% end %>
```
once this is changed to a boolean, simply:
```
no-appendfsync-on-rewrite <%= p('redis.no-appendfsync-on-rewrite') %> ? "yes" : "no" %>
```

notably, also drop the erb value quotes.

Commit:
Requested doc change

Commit:
1) no-appendfsync-on-rewrite changed to a boolean instead of the string "yes" and "no"
2) removed value quoting around auto-aof-rewrite-percentage and auto-aof-rewrite-min-size

Commit:
Deal with boolean values that can be false and you want to generate a
parameter.

norman-abramovitz commented:
@thomasmitchell  Ready for your next review.
